apollo-compiler/src/test/graphql/com/example/input_object_object_nameclash/type/Login.java

```diff
@@ -0,0 +1,88 @@
+// AUTO-GENERATED FILE. DO NOT MODIFY.
+//
+// This class was automatically generated by Apollo GraphQL plugin from the GraphQL queries it found.
+// It should not be modified by hand.
+//
+package com.example.input_object_object_nameclash.type;
+
+import com.apollographql.apollo.api.InputType;
+import com.apollographql.apollo.api.internal.InputFieldMarshaller;
+import com.apollographql.apollo.api.internal.InputFieldWriter;
+import com.apollographql.apollo.api.internal.Utils;
+import java.io.IOException;
+import java.lang.Object;
+import java.lang.Override;
+import java.lang.String;
+import org.jetbrains.annotations.NotNull;
+
+public final class Login implements InputType {
+  private final @NotNull String name;
```

Comment:
Did we miss this case? Shouldn't it generate:
```
@NotNull
private final String name;
```
instead?

Reply:
It looks like the problem only happens right now because there can be `Objects` and `InputObjects` with the same name and javapoet has to pull the package name in front of the annotation to avoid a nameclash. 
Since this is potentially a big change to change annotations everywhere,  I tried to be conservative and made the minimal patch to make the test case pass but it's certainly better to use the same logic everywhere. I'll look into that.